server/datastore/mysql/software_titles.go

```diff
@@ -104,11 +109,21 @@ func (ds *Datastore) ListSoftwareTitles(
 	// the application logic. This is because we need to support MySQL 5.7
 	// and there's no good way to do an aggregation that builds a structure
 	// (like a JSON) object for nested arrays.
-	var teamID uint
-	if opt.TeamID != nil {
-		teamID = *opt.TeamID
-	}
-	getVersionsStmt, args, err := selectSoftwareVersionsSQL(titleIDs, teamID, false)
+	getVersionsStmt, args, err := ds.selectSoftwareVersionsSQL(
+		titleIDs,
+		fleet.TeamFilter{
+			TeamID: opt.TeamID,
+			// make up a user, this function only cares about
+			// returning software versions that belong to the
+			// specified opt.TeamID, and it shouldn't include any
+			// additional versions from other teams from which the
+			// calling user might belong to.
+			User: &fleet.User{
```

Comment:
Nit: Should we generate this ad hoc team filter at the service layer?

Reply:
@gillespi314 sorry, I'm just seeing this. Are you thinking of another parameter for this function? the tricky thing is that this function already receives a team filter, but it shouldn't be passed to `selectSoftwareVersionsSQL`, because otherwise it might include versions that the user can see but don't belong to the team, due to the way this filters work.

Reply:
I was just wondering if we could add the dummy admin team filter in `svc.ListSoftwareTitles` right before it calls `ds.ListSoftwareTitles` so we keep auth-related stuff at the service level as much as possible.

Reply:
@gillespi314 that's a good question, the thing I'm concerned about is that by doing that now any caller of this function needs to know about the inner implementation detail noted in the comment, so it's a tough one 🤔

Reply:
@roperzh, shouldn't we be pulling the user from the viewer context in `svc.ListSoftwareTitles` and using that to populate the user filter? In which case, I don't think we'd need to make up a dummy user in the datastore layer.

Reply:
@gillespi314 `whereFilterGlobalOrTeamIDByTeams` does this:

https://github.com/fleetdm/fleet/blob/a371e9d0f0add6d0011b2b5e3177ae3c05eecccf/server/datastore/mysql/mysql.go#L1006

which will all the users' team to the clause. In this specific case we only want to add the software versions that belong to the specific team we want.

so, I don't think we should be passing anything from the caller (the caller is already passing a team ID and that's all it should care about), but we could consider refactoring `selectSoftwareVersionsSQL`? or defining another method?

Reply:
I see. So what we're after in this particular case is the way the prior version of `selectSoftwareVersionsSQL` worked, where it just looked at the passed team ID without reference to the user?

Reply:
Yeah, I think that's probably the best place to address it. What if we change the `filter.User == nil` exit to return the filter for just the specified team id (or 0)?

Reply:
thanks! you're right!